api/controller.py

```diff
@@ -843,7 +854,13 @@ def container(self):
                                '<http://www.w3.org/TR/annotation-protocol/>; rel="http://www.w3.org/ns/ldp#constrainedBy"']
             headers['Content-Type'] = AnnotationWriter.CONTENT_TYPE
 
-            container = AnnotationWriter.annotation_container_for(patron)
+            container, timestamp = AnnotationWriter.annotation_container_for(patron)
+            etag = 'W/""'
+            if timestamp:
+                etag = 'W/"%s"' % timestamp
+                headers['Last-Modified'] = timestamp
```

Comment:
Is this timestamp going to be in the right format? I'm not sure and it's probably worth explicitly formatting the datetime object as a string. Search https://www.ietf.org/rfc/rfc2616.txt for "rfc1123-date" for the valid formats.